3761/1/gem_statistic_fetcher.rb

```diff
@@ -0,0 +1,33 @@
+require './page.rb'
+class GemStatisticFetcher
+  attr_reader :gem_name, :agent
+  attr_reader :stats
+
+  def initialize(gem_name, agent)
+    @gem_name = gem_name
+    @agent = agent
+    @stats = {}
+  end
+
+  def call
+    page_gem_in_ribygems = Page.new('https://rubygems.org/gems/' + gem_name, agent)
+    return if page_gem_in_ribygems.data == false
+    @stats = collect_statistic(page_gem_in_ribygems)
+    @stats = collect_used_by(page_gem_in_ribygems)
```

Comment:
You don't need to assign at all, and your code will fell down if you swap this calls